server/src/main/java/org/elasticsearch/ingest/IngestStats.java

```diff
@@ -135,6 +136,21 @@ public Stats getTotalStats() {
         return processorStats;
     }
 
+    @Override
```

Comment:
@jakelandis I added `equals` and `hashCode` objects here as I needed them for testing and for hashing the objects. 

Adding them seemed like a no brainer to me, but I don't want to change this code without somebody who works on ingest giving their blessing.

Reply:
this part of the change LGTM (didn't review the other parts)